packages/components/src/internal/util/Date.ts

```diff
@@ -398,7 +407,11 @@ export function parseDate(
  */
 export function parseTime(time: string): Date {
     if (!time) return null;
-    let valueFormat = includesSeconds(time) ? ISO_TIME_FORMAT_STRING : ISO_SHORT_TIME_FORMAT_STRING;
+    let valueFormat = includesMilliSeconds(time)
```

Comment:
Two things.

1. Add a runtime check for string type.
2. nit: Could be made more readible:

```ts
export function parseTime(time: string): Date | null {
    if (!time || typeof time !== 'string') return null;
    let timeFormat: string;

    if (includesMilliSeconds(time)) {
        timeFormat = ISO_LONG_TIME_FORMAT_STRING;
    } else if (includesSeconds(time)) {
        timeFormat = ISO_TIME_FORMAT_STRING;
    } else {
        timeFormat = ISO_SHORT_TIME_FORMAT_STRING;
    }

    if (includesAMPM(time)) {
        timeFormat = timeFormat.replace('HH', 'hh');
        timeFormat += ' a';
    }

    // https://stackoverflow.com/a/68727535
    const date = safeParse(time, timeFormat, new Date());
    return isValid(date) ? date : null;
}
```